core/state/commontrie/trie_test.go

```diff
@@ -0,0 +1,41 @@
+package commontrie
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTrieAdapter(t *testing.T) {
```

Comment:
This comment is for another PR. To guarantee that the old and the new trie will produce the same behaviour, we should write a single set of tests that depends on this interface, and run them with both the old and the new trie. 

The reason why I suggest to avoid writing 2 separate test suites for 2 tries is because of the scenarios where a single typo in 1 of the 2 tests can make both tests passed even though they produce different behaviours.